### PR TITLE
Create codeql.yml

### PR DESCRIPTION
**Pull request by zhangchl007**
code scan

### CODE REVIEW

**Comment by zhangchl007** [comment]
@zhangchl168
We add the workflow of code scan, Please review the request,

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    #   modify them (or add more) to build your code if your project, please refer to the EXAMPLE below for guidance.
+
+    # - run: |
+    #     echo "Run, Build Application using script"
+    #     ./location_of_script_within_repo/buildscript.sh
+
+    - name: Perform CodeQL Analysis
+      uses: github/codeql-action/analyze@v2
+      with:
+        category: "/language:${{matrix.language}}"
```
Comment (zhangchl168) [inline .github/workflows/codeql.yml:77]: @zhangchl007 
please make sure it's correct
```suggestion
        category: "/language:${{matrix.language}}"
```

**Review by zhangchl168 — COMMENTED** [review]
please check the comment

File: .github/workflows/codeql.yml (hunk 1 above)
Comment (zhangchl007) [reply .github/workflows/codeql.yml:77]: corrected it, 

1. [make it as the regular check in the future ]

**Review by zhangchl168 — APPROVED** [review]
now it's a clean code

**Pull request merged**